recipes-app/iot2050-event-record/WATCHDOG.md

```diff
@@ -0,0 +1,81 @@
+Due to the fact that the watchdog is non-stoppable, getting a watchdog
+event always requires opening the watchdog and feeding the watchdog. So
+the watchdog event could not be included into the `iot2050-event-record`
+service.
+
+This README file explains how to get the watchdog reset status and how to
+inject it into `iot2050-event-record` service.
+
+# How to get the watchdog reset status?
+
+The `wdt_example.py` below shows how to get the watchdog reset status.
+
+```py
+import array
+import fcntl
+import os
+import psutil
+import time
+from datetime import datetime
+
+# Implement _IOR function for wdt kernel ioctl function
+_IOC_NRBITS   =  8
+_IOC_TYPEBITS =  8
+_IOC_SIZEBITS = 14
+_IOC_DIRBITS  =  2
+
+_IOC_NRSHIFT = 0
+_IOC_TYPESHIFT =(_IOC_NRSHIFT+_IOC_NRBITS)
+_IOC_SIZESHIFT =(_IOC_TYPESHIFT+_IOC_TYPEBITS)
+_IOC_DIRSHIFT  =(_IOC_SIZESHIFT+_IOC_SIZEBITS)
+
+_IOC_NONE = 0
+_IOC_WRITE = 1
+_IOC_READ = 2
+def _IOC(direction,type,nr,size):
+    return (((direction)  << _IOC_DIRSHIFT) |
+        ((type) << _IOC_TYPESHIFT) |
+        ((nr)   << _IOC_NRSHIFT) |
+        ((size) << _IOC_SIZESHIFT))
+def _IOR(type, number, size):
+    return _IOC(_IOC_READ, type, number, size)
+
+WDIOC_GETBOOTSTATUS = _IOR(ord('W'), 2, 4)
+WDIOF_CARDRESET = 0x20
+WDT_PATH = "/dev/watchdog"
+
+EVENT_STRINGS = {
+    "wdt": "{} watchdog reset is detected",
+    "no-wdt": "{} watchdog reset isn't detected"
+}
+
+def feeding_the_watchdog(fd):
+    while True:
+       ret = os.write(fd, b'watchdog')
+       print("Feeding the watchdog ...")
+       # Let's say the watchdog timeout is more than 30 s
+       time.sleep(30)
```

Comment:
What if systemd is feeding the watchdog? Do we still need this? Can we actually still use all this?

Reply:
Rewrote the example to get boot status via `/sys/class/watchdog/watchdog0/bootstatus`.